posthog/hogql/test/test_metadata.py

```diff
@@ -105,3 +108,61 @@ def test_metadata_expr_resolve_error(self):
                 ],
             },
         )
+
+    def test_metadata_in_cohort(self):
+        cohort = Cohort.objects.create(team=self.team, name="cohort_name")
+        query = (
+            f"select person_id from events where person_id in cohort {cohort.pk} or person_id in cohort '{cohort.name}'"
+        )
+        metadata = self._select(query)
+        self.assertEqual(
+            metadata.dict(),
+            metadata.dict()
+            | {
+                "isValid": True,
+                "inputExpr": None,
+                "inputSelect": query,
+                "notices": [
+                    {
+                        "message": f"Cohort #{cohort.pk} can also be specified as '{cohort.name}'",
+                        "start": 55,
+                        "end": 55 + len(str(cohort.pk)),
+                        "fix": f"'{cohort.name}'",
```

Comment:
I would think a fix should be one way, and this "fix" can be "fixed" by applying the other one ('Replace with numeric ID"), which can go on forever. 🤔 It's very useful to show the cohort name in a tooltip, but I suggest we don't have replacing that with the name as a fix, given the potential problems with renaming
BTW all the other notices end with a period, and this one doesn't. Given these are kind of linter hints, I would find them most natural if they were all single-sentence without a period

Reply:
- I changed a bunch of notices and smaller errors (that'll also end up in notices/errors here) to remove periods in the end.
- Why I included both actions: I think it's not right to assume what the user wants. They might actually want to use strings, as they have some system of naming cohorts they're using to make their queries easier to write/understand. I don't know and can't assume, but know I'd get frustrated when I'd have to copy/paste the text from the tooltip into the editor. So I left it in 🤷